### PR TITLE
[Network] BREAKING CHANGE: `az network vnet subnet create`: Disable `PrivateEndpointNetworkPolicies` by default

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network vnet subnet create

**Description**

> Depending on that we may need a fix to default this to “Disabled” (we are discussing and closing on it).

Doc: https://docs.microsoft.com/en-us/azure/private-link/disable-private-endpoint-network-policy
Swagger: https://github.com/Azure/azure-rest-api-specs/pull/19577

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

**Comment by avijitgupta** [comment]
do you need to change def update_subnet as well?
If customer specifies a none, what happens in update_subnet?

**Comment by necusjz** [comment]
> do you need to change def update_subnet as well? If customer specifies a none, what happens in update_subnet?

When it is None, it will keep the previous status, i.e., no any updates, unless explicitly declared to be True or False.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -7119,7 +7120,7 @@ def create_subnet(cmd, resource_group_name, virtual_network_name, subnet_name,
     if delegations:
         subnet.delegations = delegations
 
-    if disable_private_endpoint_network_policies is True:
+    if disable_private_endpoint_network_policies is None or disable_private_endpoint_network_policies is True:
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7123]: Why not change the function signature
```
disable_private_endpoint_network_policies=True,
```

Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:7123]: Did some research, the reason why default argument "True" becomes "1", it's a kind of distinguishing:
https://github.com/microsoft/knack/blob/b4d9e66b469ef7c18a1f228f803505b7f550bdc1/knack/commands.py#L122-L124

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**